### ISSUE TITLE
JoinDomainOrWorkGroup returns 1332

### ISSUE THREAD

**Issue by joeypiccola** [trlinkin/puppet-domain_membership#11]
Hi @trlinkin 

I can't seem to figure this one out. Simply trying to join the machine to a domain. 

Thanks, 
Joey

`
class nameservers::djoin (

  $domain     = $nameservers::params::domain,
  $machine_ou = $nameservers::params::machine_ou,
  $username   = $nameservers::params::username,
  $password   = $nameservers::params::password,

) inherits nameservers::params {

  class { '::domain_membership':
    domain       => $domain,
    machine_ou   => $machine_ou,
    username     => $username,
    password     => $password,
  }
  
}
`

Output. 

`
==> default: Notice: Compiled catalog for vagrant-3vvbarv.testdomain.com in environment production in 1.44 seconds
==> default: Info: Applying configuration version '1475098551'
==> default: Error: exit (Get-WmiObject -Class Win32_ComputerSystem).JoinDomainOrWorkGroup('my.domain.com','myPassword','svc_djoin@my.domain.com','OU=Servers,DC=my,DC=domain,DC=com',1).ReturnValue returned 1332 instead of one of [0]
==> default: 
==> default: Error: /Stage[main]/Domain_membership/Exec[join_domain]/returns: change from notrun to 0 failed: exit (Get-WmiObject -Class Win32_ComputerSystem).JoinDomainOrWorkGroup('my.domain.com','myPassword','svc_djoin@my.domain.com','OU=Servers,DC=my,DC=domain,DC=com',1).ReturnValue returned 1332 instead of one of [0]
`

**Comment by trlinkin** [comment]
TL;DR - Set "join_options" to "3"

Under the hood I'm calling some fun WMI functionality. How that part works is specified in this wonderful (SEE: dense) bit of documentation from Microsoft: https://msdn.microsoft.com/en-us/library/aa392154(v=vs.85).aspx

**Comment by joeypiccola** [comment]
well, that fixed it. i understand now too, you're NETSETUP_JOIN_DOMAIN (0x00000001) + NETSETUP_ACCT_CREATE (0x00000002) = 0x3. maybe update the documentation with a mention that if you use the default 0x1 it's expecting the computer object be pre-staged? 

@trlinkin thanks for the quick explanation and response!

**Comment by trlinkin** [comment]
Yeah! I'll clean up the docs a bit, glad it worked!
